### DISCUSSION TITLE
How to use parameters  in the catalog file

### DISCUSSION THREAD

**Discussion by babuas06** [kedro-org/kedro#1326] [Idea]
Hello Guys, 
I want to use values from parameters in the data catalog. I am using TemplatedConfigLoader.
for example, I want to dynamycally pull values for file path.
filepath: "abfs://container_name/${value_from_parameters}.csv"
I am not sure how to get the value_from_parameters into catalog. 

Thank you very much for your help.

**Comment by datajoely** [comment]
When you say parameters - do you mean via the CLI?

Out of the box you need to put the key in `globals.yml` for it to be picked up.

**Reply by babuas06** [reply]
I want to pass the parameters via the CLI.

**Reply by babuas06** [reply]
Actually, my requirement is to pull some values which are available in parameters.yml and one value is passed from CLI. 
for example filepath: "abfs://container-name/${value_from_parameter}/${value_from_parameter}/${value_from_CLI}.csv"
Can you please give an example of how to achieve this?
Thanks

**Reply by datajoely** [reply]
Hi @babuas06 so I'll address this in two parts. I should also say because Kedro is designed around reproducible runs this sort of ephemeral configuration (especially via the CLI) isn't always encouraged.

(1) Value from parameter

You need to define the value in `globals.yml` and then inject it into your parameters and the catalog entry from the same source. You can't go parameters -> catalog directly, they're designed to be independent.

(2) Value from CLI

This is a little tricker you need to introduce your own run command within your project. To do this you can copy and paste the implementation in from `kedro.framework.cli.project.run` into a project file called `cli.py` you can then intercept the parameters and pass them as `extra_params` which can be configured to update the `globals` dict and are thus usable within `TemplatedConfigLoader`

**Reply by babuas06** [reply]
Ok Thanks for the suggestion.
